configdialog.cpp

```diff
@@ -518,6 +535,271 @@ void PaletteCustomizeDialog::get_and_save_color(int palette_index) {
 	}
 }
 
+std::wstring PaletteCustomizeDialog::export_palettes() {
+	auto all_palettes = m_palette_repo.get_all_custom_palettes();
+	std::wstring palettes_string = L"";
+
+	for (const auto &palette : all_palettes) {
+		auto name = palette.name;
+		auto *data = palette.data;
+
+		palettes_string.append(palette.name + L": ");
+		std::wstring colors = m_palette_repo.serialize(*data) + L";";
+		// Add extra "fake" colors so that a palette definition is guaranteed to be longer than the maximum allowed name length
+		while (colors.length() < PaletteCustomizeDialog::MaxPaletteNameSize) {
+			colors.append(L"#000000;");
+		}
+		palettes_string.append(colors);
+		palettes_string.append(L"\r\n");
+	}
+	palettes_string.append(L"~");
+
+	return palettes_string;
+}
+
+void PaletteCustomizeDialog::import_palettes(std::wstring *palettes) {
+	using namespace std;
+
+	const static wstring valid_chars = L"qwertyuiopasdfghjklzxcvbnmQWERTYUIOPASDFGHJKLZXCVBNM0123456789-_!?:";
+	const static wstring valid_hex_chars = L"0123456789abcdefABCDEF";
+	const static size_t colors_amount = max(7, (size_t)ceil(PaletteCustomizeDialog::MaxPaletteNameSize / 8.0));
```

Comment:
Why is this ever not 7?

Reply:
originally it was for extra validation, but I've decided that it's completely unnecessary.